File: project_euler/Problem100/Problem100.js
function Problem100() {
    let target = 1000000000000
    let index = 1
    let result = 1
    while (index < target) {
        let auxiliary = index * 2 + result * 3 - 2
        index = index * 3 + result * 4 - 3
        result = auxiliary
    }
    console.log(result)
}

module.exports = {
    Problem100 : Problem100
};

File: project_euler/Problem100/Problem100.java
package project_euler;

public class Problem100 {
    public void solution() {
        long target = 1000000000000L;
        long index = 1L;
        long result = 1L;
        while (index < target) {
            long auxiliary = (index << 1) + result * 3 - 2;
            index = index * 3 + (result << 2) - 3;
            result = auxiliary;
        }
        System.out.println(result);
    }
}
